tests/dhcp_server/test_dhcp_server_stress.py

```diff
@@ -0,0 +1,202 @@
+    vlan_net = ipaddress.ip_network(address=first_vlan_prefix, strict=False)
+    vlan_gateway = first_vlan_prefix.split('/')[0]
+    vlan_hosts = [str(host) for host in vlan_net.hosts()]
+    # to avoid configurate an range contains gateway ip, simply ignore all ip before gateway and gateway itself
+    vlan_hosts_after_gateway = vlan_hosts[vlan_hosts.index(vlan_gateway) + 1:]
+    pytest_assert(len(vlan_hosts_after_gateway) >= 2, 'Vlan size is too small for testing')
+    vlan_setting = {
+        'vlan_name': first_vlan_name,
+        'vlan_gateway': vlan_gateway,
+        'vlan_subnet_mask': str(vlan_net.netmask),
+        'vlan_hosts': vlan_hosts_after_gateway,
+        'vlan_member_with_ptf_idx': vlan_member_with_ptf_idx,
+    }
+
+    logging.info("The vlan_setting before test is %s" % vlan_setting)
+    return vlan_setting['vlan_name'], \
+        vlan_setting['vlan_gateway'], \
+        vlan_setting['vlan_subnet_mask'], \
+        vlan_setting['vlan_hosts'], \
+        vlan_setting['vlan_member_with_ptf_idx']
+
+
+def test_dhcp_server_with_multiple_dhcp_clients(
+    duthost,
+    ptfhost,
+    ptfadapter,
+    parse_vlan_setting_from_running_config
+):
+    """
+        Make sure all ports can get assigend ip when all ports request ip at same time
+    """
+    vlan_name, gateway, net_mask, vlan_hosts, vlan_members_with_ptf_idx = parse_vlan_setting_from_running_config
+    start_command = " && ".join(["dhclient -nw eth%s" % ptf_index for _, ptf_index in vlan_members_with_ptf_idx])
+    end_command = " ; ".join(["dhclient -r eth%s" % ptf_index for _, ptf_index in vlan_members_with_ptf_idx])
+    try:
+        config_to_apply = empty_config_patch()
+        dut_ports, _ = zip(*vlan_members_with_ptf_idx)
+        exp_assigned_ip_ranges = [[ip] for ip in vlan_hosts[:len(vlan_members_with_ptf_idx)]]
+        append_common_config_patch(
+            config_to_apply,
+            vlan_name,
+            gateway,
+            net_mask,
+            dut_ports,
+            exp_assigned_ip_ranges
+        )
+        apply_dhcp_server_config_gcu(duthost, config_to_apply)
+        ptfhost.shell(start_command)
+
+        def all_ip_shown_up(ptfhost, expected_assigned_ips):
+            ip_addr_output = ptfhost.shell("ip addr")['stdout']
+            for expected_ip in expected_assigned_ips:
+                if expected_ip not in ip_addr_output:
+                    return False
+            return True
+        expected_assigned_ips = [range[0] for range in exp_assigned_ip_ranges]
+        pytest_assert(
+            wait_until(120, 1, 1,
+                       all_ip_shown_up,
+                       ptfhost,
+                       expected_assigned_ips),
+            'Not all configurated IP shown up on ptf interfaces'
+        )
+    finally:
+        ptfhost.shell(end_command, module_ignore_errors=True)
+        ptfhost.shell("killall dhclient", module_ignore_errors=True)
+
+
+def test_dhcp_server_with_large_number_of_discover(
```

Comment:
Parametrize packet count and tolerant time

Reply:
Added the parameter